src/components/QuantityInput.tsx

```diff
@@ -0,0 +1,75 @@
+import { Input, InputProps, Text } from "@saleor/macaw-ui-next";
+import React, { ReactNode } from "react";
+import { FormattedMessage } from "react-intl";
+
+interface QuantityInputProps extends InputProps {
+  disabled?: boolean;
+  className?: string;
+  value: number;
+  max: number;
+  isError?: boolean;
+  labels?: {
+    outOf?: ReactNode;
+    error?: string;
+  };
+}
+
+export const QuantityInput: React.FC<QuantityInputProps> = ({
+  disabled,
+  className,
+  value,
+  onChange,
+  max,
+  isError,
+  labels,
+  ...props
+}) => (
+  <Input
+    {...props}
+    disabled={disabled}
+    className={className}
+    type="number"
+    __minWidth="40px"
+    width="100%"
+    value={value}
+    onChange={onChange}
+    endAdornment={
+      <Text minWidth={10} paddingRight={2} __marginLeft="-14px">
+        {labels?.outOf ?? (
+          <FormattedMessage
+            id="CWDuMj"
+            defaultMessage="/ {max}"
+            description="remaining quantity"
+            values={{
+              max,
+            }}
+          />
+        )}
+      </Text>
+    }
+    max={max}
+    min={0}
+    textAlign="right"
+    error={isError}
+    helperText={
+      isError && (
+        <Text
+          color="critical1"
+          __marginTop="-2px"
+          display="block"
+          position="absolute"
+          bottom={0}
+          fontSize={1}
+        >
+          {labels?.error ?? (
```

Comment:
question: shouldn't we show label.error value that has error description instead of a generic error message?

Reply:
Not sure I understand, I used `??` so if there's an error label then it will show, otherwise it's going to be a generic error message

Reply:
oh, ok sorry you're right 👍🏻